### PR TITLE
usergroup PUT 

### PR DESCRIPTION
**Pull request by phoenixAja**
### Description

adding put request to usergroup api endpoint that allows someone to update User entry

#### Issue
[ch134917](https://app.clubhouse.io/genepi/story/134917)

### Test plan

wrote some tests

### CODE REVIEW

**Comment by phoenixAja** [comment]
@ttung can you help me? my test is failing, i think it has something to do with syncing the sessions? Or do you know what I'm doing wrong? I checked that the view PUT option worked manually with pdb and the tos field gets updated, just in the test the session seems like it's not getting updated?

File: src/backend/aspen/app/views/usergroup.py [hunk 1]
```diff
@@ -1,17 +1,40 @@
...
+                "user": user.to_dict(),
+                "group": user.group.to_dict(),
+            }
+            return jsonify(user_groups)
+
+        if request.method == "PUT":
+            fields_to_update: Dict[str, Union[str, bool]] = json.loads(
+                request.get_json()
+            )
+            for key, value in fields_to_update.items():
```
Comment (tihuan) [inline src/backend/aspen/app/views/usergroup.py:33]: Should we restrict what fields a user is allowed to update using an allow list? Or are the fields here pretty harmless even if they change things unexpectedly? For example, if they change group id to something else, would they gain access to something they shouldn't? Thank you!

Comment (phoenixAja) [reply src/backend/aspen/app/views/usergroup.py:33]: oh that's a good idea!! thanks!

Comment (phoenixAja) [reply src/backend/aspen/app/views/usergroup.py:33]: @tihuan just added this, let me know if you think that solves it!

Comment (tihuan) [reply src/backend/aspen/app/views/usergroup.py:33]: Awesomeee! Thanks so much for the quick update, @phoenixAja 🎉 ! LGTM!!

**Review by tihuan — APPROVED** [review]
LGTM!! 🚀 🚀 🚀

File: src/backend/aspen/app/views/tests/test_views.py [hunk 2]
```diff
@@ -15,6 +16,41 @@ def test_usergroup_view(session, app, client):
...
+    session.commit()
+    with client.session_transaction() as sess:
+        sess["profile"] = {"name": user.name, "user_id": user.auth0_user_id}
+
+    data = {"agreed_to_tos": True}
+    res = client.put("/api/usergroup", json=json.dumps(data))
+
+    # start a new transaction
+    session.close()
+    session.begin()
```
Comment (ttung) [inline src/backend/aspen/app/views/tests/test_views.py:32]: I suspect `session.commit()` is enough to force a new transaction to start.

File: src/backend/aspen/app/views/usergroup.py [hunk 3]
```diff
@@ -1,17 +1,44 @@
...
+from flask import jsonify, request, Response, session
 
 from aspen.app.app import application, requires_auth
 from aspen.app.views.api_utils import get_usergroup_query
 from aspen.database.connection import session_scope
+from aspen.database.models.usergroup import User
 
 
-@application.route("/api/usergroup", methods=["GET"])
+@application.route("/api/usergroup", methods=["GET", "PUT"])
```
Comment (ttung) [inline src/backend/aspen/app/views/usergroup.py:12]: I think it probably makes sense to make PUT a separate route from GET.  They don't have enough in commonality to justify a single route.

Comment (phoenixAja) [reply src/backend/aspen/app/views/usergroup.py:12]: really you think? I thought it was cleaner to have (very basic) get, put, and post in the same view. I don't mind splitting it out though if you think that would be better.

File: src/backend/aspen/app/views/tests/test_views.py (hunk 2 above)
Comment (phoenixAja) [reply src/backend/aspen/app/views/tests/test_views.py:32]: interesting, i guess it would

**Pull request merged**